### PR TITLE
Fix deprecation warning with asdf 2.8

### PR DESCRIPTION
**Pull request by eslavich**
### Description

This pull request is to address a deprecation warning in a test that will be caused by asdf 2.8 (due to be released later today).

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -472,6 +472,8 @@ astropy.io.misc
 - Fix ASDF serialization of circular model inverses, and remove explicit calls
   to ``asdf.yamlutil`` functions that became unnecessary in asdf 2.6.0. [#10189, #10384]
 
+- Resolve deprecation warning from asdf 2.8. [#11738]
```
Comment (pllim) [inline CHANGES.rst:475]: We have switched to towncrier. Please see https://github.com/astropy/astropy/blob/main/docs/changes/README.rst and squash this out. Thanks!

Comment (pllim) [reply CHANGES.rst:475]: p.s. Not even sure if change log is needed in this case.

Comment (eslavich) [reply CHANGES.rst:475]: Oh oops, I knew that too.  I'll just remove it, I agree it's not a particularly interesting detail.

File: astropy/io/misc/asdf/tags/table/tests/test_table.py [hunk 2]
```diff
@@ -98,8 +98,15 @@ def test_table_inline(tmpdir):
     def check(ff):
         assert len(list(ff.blocks.internal_blocks)) == 0
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check,
-                                  write_options={'auto_inline': 64})
+    if hasattr(asdf, 'config_context'):
```
Comment (pllim) [inline astropy/io/misc/asdf/tags/table/tests/test_table.py:101]: Why not use version check? That way, we can easily search for dead code when we bump:

https://github.com/astropy/astropy/blob/715d7a660a554f8880d8df82fae28180eeb21e2f/setup.cfg#L80

Also see: https://github.com/astropy/astropy/blob/main/astropy/utils/compat/optional_deps.py

**Review by pllim — APPROVED** [review]
LGTM as long as tests pass. Thanks for the quick fix!

**Comment by eslavich** [comment]
@pllim is it safe to ignore the readthedocs failure?  I see that happening on other PRs too...

**Comment by pllim** [comment]
But this one is a time out... Hmm. Anyways, I don't think it is related to your PR. Thanks!

**Comment by eteq** [comment]
turns out this is needed in 4.3 too, so backporting in a80c002

**Comment by pllim** [comment]
I moved the milestone to LTS because the lack of this is causing v4.0.x PRs to fail.

@MeeseeksDev backport to v4.0.x

**Pull request merged**